### PR TITLE
implementing package exclude/ignore files for layers packages.

### PR DESCRIPTION
**Pull request by agutoli**
https://github.com/agutoli/serverless-layers/issues/118

### CODE REVIEW

**Comment by agutoli** [comment]
@stanyq4 Would be great to receive feedbacks regarding this implementation.

File: src/package/Dependencies.js [hunk 1]
```diff
@@ -31,6 +31,46 @@ class Dependencies extends AbstractService {
...
+     * you can always re-include previously excluded files and directories.
+     *
+     * Reference: https://www.serverless.com/framework/docs/providers/aws/guide/packaging
+     */
+    for (let pattern of this.plugin.service.package.patterns) {
+      if (pattern.startsWith('!')) {
+        const resolvedFiles = await resolveFile(pattern.substr(1), {
+          cwd: this.layersPackageDir
+        });
+        filesToIgnore = filesToIgnore.concat(resolvedFiles);
```
Comment (dorman99) [inline src/package/Dependencies.js:55]: why do you, ignore the exclude one ?

**Comment by dorman99** [comment]
Screen Shot 2023-02-10 at 12 09 01

and got this message, after trying this changes. i guess, the function zip still got the modules that already excluded

**Comment by dorman99** [comment]
qq, what does the copy function do ?

**Comment by fyelci** [comment]
I have this problem as well. If this PR solves the problem, can someone please merge it?

**Comment by jaywala** [comment]
Is there a plan to merge this PR? I have this issue too with aws-sdk v3 modules that are already included on node18 that I need to exclude

File: src/package/Dependencies.js (hunk 1 above)
Comment (agutoli) [reply src/package/Dependencies.js:55]: This plugin doesn't read the `serverless.yaml` file when doing `npm install ` to create the aws lambda layer. Thats why I have to check what files is being ignored to be able to remove unnecessary files.

**Pull request merged**